doc/dev/CustomRepoGuide.md

```diff
@@ -0,0 +1,387 @@
+
+So the header is quiet self explanatory:
+
+~~~yml
+language: cpp
+sudo: false
+services: docker
+dist: trusty
+~~~
+
+We want to test something written in cpp, do not need sudo permissions on the server and request the additional docker (virtual machine) service on a ubuntu trusty (14.04) system.
+
+The next part sets your different test configurations
+
+~~~yml
+matrix:
+    include:
+        - compiler: clang
+          env: BUILD_TYPE=Release
+        - compiler: g++
+          env: BUILD_TYPE=Release
+        - compiler: g++
+          env: BUILD_TYPE=Debug COVERAGE=--coverage
+        - compiler: clang
+          env: BUILD_TYPE=Release WERROR=-Werror
+        - compiler: g++
+          env: BUILD_TYPE=Release WERROR=-Werror
+    allow_failures:
+        - env: BUILD_TYPE=Release WERROR=-Werror
+~~~
+
+The `include:` section covers the different build types you want to use. The 'compiler:' flag is obvious. The `env:` flag sets environmental variables that you use later inside instructions and if statements. The `allow_failures:` section specifies build setups, that are allowed to fail. So the rating if your test passed or not does not depend on this setup. In the example above, we want two builds to fail, if the compiler creates any warnings, but we don't want those builds to cause a failure of the whole test series. 
+
+Now we define what we want to do before the tests are executed using the flag `before_install:`. (Notice that all the following sections basically just take bash commands that you can run on your own system.)
+
+~~~yml
+before_install:
+    - travis_retry timeout 120 docker pull nuto/nuto_docker:dev
+    - docker run -itd --name dock -u 0 -v $(pwd):/home/nuto/source nuto/nuto_docker:dev
+~~~
+
+First we try to pull the current NuTo docker image (more infos regarding docker can be found in the following section):
+
+~~~bash
+docker pull nuto/nuto_docker:dev
+~~~
+
+The preceding `travis_retry timeout 120` just tells travis to retry it after 120 second in case the server is not responding. Then we start the image with 
+
+~~~bash
+docker run -itd --name dock -u 0 -v $(pwd):/home/nuto/source nuto/nuto_docker:dev
+~~~
+
+This will start the image (`docker run  ... nuto/nuto_docker:dev`) in detached mode (`-itd`). In detached mode, the image won't "take over" the bash on travis. To send commands to that image, we give it the name dock (`--name dock`). The command `-v $(pwd):/home/nuto/source` mounts our current working directory (`$(pwd)`) on travis to the folder `/home/nuto/source` inside our image. This way we are capable of sharing data between the host (Travis) and our virtual machine. In this case we share the NuTo source code, which Travis automatically pulls, with our image. This way we avoid pulling it twice. The flag `-u 0` tells the image, that we want to log in as root. At this point this is not necessary, but if you want write access to the mounted directory (see codecov section) you need active root permissions.
```

Comment:
- "image -> container", "virtual machine -> container"
- we do the codecov stuff on NuTo without `-u 0`?!